web3/_utils/abi.py

```diff
@@ -51,14 +51,44 @@ def get_abi_input_types(abi):
     if 'inputs' not in abi and abi['type'] == 'fallback':
         return []
     else:
-        return [arg['type'] for arg in abi['inputs']]
+        return [collapse_if_tuple(abi_input) for abi_input in abi['inputs']]
+
+
+def collapse_if_tuple(abi):
```

Comment:
This function should be removed from here, and it's callers should use the one in `eth-utils` instead.  But, [that change in `eth-utils`](https://github.com/ethereum/eth-utils/pull/141/commits/227beca013b4167a148e610d238892d9679f0e59) hasn't been released yet...